src/drivers/pca9685_pwm/PCA9685.h

```diff
@@ -0,0 +1,168 @@
+#define PCA9685_REG_LED0_OFF_L 0x08		// LED0 output and brightness control byte 2
+#define PCA9685_REG_LED0_OFF_H 0x09		// LED0 output and brightness control byte 3
+#define PCA9685_REG_LED_INCREMENT 4 	// compute the other 15 channels
+#define PCA9685_REG_ALLLED_ON_L 0xFA    // load all the LEDn_ON registers, byte 0 (turn 0-7 channels on)
+#define PCA9685_REG_ALLLED_ON_H 0xFB	// load all the LEDn_ON registers, byte 1 (turn 8-15 channels on)
+#define PCA9685_REG_ALLLED_OFF_L 0xFC	// load all the LEDn_OFF registers, byte 0 (turn 0-7 channels off)
+#define PCA9685_REG_ALLLED_OFF_H 0xFD	// load all the LEDn_OFF registers, byte 1 (turn 8-15 channels off)
+#define PCA9685_REG_PRE_SCALE 0xFE		// prescaler for output frequency
+#define PCA9685_REG_TESTMODE 0xFF       // test mode register
+
+// Mode register 1
+#define PCA9685_MODE1_RESTART_MASK 0x80
+#define PCA9685_MODE1_EXTCLK_MASK 0x40
+#define PCA9685_MODE1_AI_MASK 0x20
+#define PCA9685_MODE1_SLEEP_MASK 0x10
+#define PCA9685_MODE1_SUB1_MASK 0x08
+#define PCA9685_MODE1_SUB2_MASK 0x04
+#define PCA9685_MODE1_SUB3_MASK 0x02
+#define PCA9685_MODE1_ALLCALL_MASK 0x01
+
+// Mode register 2
+#define PCA9685_MODE2_INVRT_MASK 0x10
+#define PCA9685_MODE2_OCH_MASK 0x08
+#define PCA9685_MODE2_OUTDRV_MASK 0x04
+#define PCA9685_MODE2_OUTNE_MASK 0x03
+
+// LED_ON, LED_OFF control registers
+#define PCA9685_LED0_ON_L_MASK 0xFF
+#define PCA9685_LED_ON_FULL_ON_OFF_MASK 0x10
+#define PCA9685_LED0_ON_H_MASK 0x0F
+
+// PRE_SCALE register
+#define PCA9685_PRE_SCALE_MASK 0xFF
+
+#define PCA9685_PWM_CHANNEL_COUNT 16
+#define PCA9685_PWM_RES 4096        //Resolution 4096=12bit
+#define PCA9685_CLOCK_INT 25000000.0 //25MHz internal clock
+#ifndef PCA9685_CLOCL_EXT
+#define PCA9685_CLOCK_FREQ PCA9685_CLOCK_INT   // use int clk
+#else
+#define PCA9685_CLOCK_FREQ PCA9685_CLOCK_EXT   // use ext clk
+#endif
+
+#define PWM_DEFAULT_FREQUENCY 50    // default pwm frequency
+
+//! Main class that exports features for PCA9685 chip
+class PCA9685 : public device::I2C
+{
+public:
+	PCA9685(int bus, int addr);
+
+	int Start();
+
+	int Stop();
+
+	/*
+	 * outputs formatted to us.
+	 */
+	int updatePWM(const uint16_t *outputs, unsigned num_outputs);
+
+	int setFreq(int freq);
+
+	~PCA9685() override = default;
+
+	int init() override;
+
+protected:
+	int probe() override;
+
+	static const uint8_t DEFAULT_MODE1_CFG = 0x20;
+	static const uint8_t DEFAULT_MODE2_CFG = 0x04;
+
+	double _Freq = PWM_DEFAULT_FREQUENCY;
+
+	/**
+	 * set PWM value for a channel[0,15].
+	 * value should be range of 0-4095
+	 */
+	void setPWM(uint8_t channel, const uint16_t &value);
+
+	/**
+	 * set all PWMs in a single I2C transmission.
+	 * value should be range of 0-4095
+	 */
+	void setPWM(uint8_t channel_count, const uint16_t *value);
+
+	/*
+	 * disable all of the output
+	 */
+	void disAll();
+
+	/*
+	 * set clock divider
+	 * this func has Super Cow Powers
+	 */
+	void setDivider(uint8_t value);
+
+	/*
+	 * turn off oscillator
+	 */
+	void gosleep();
```

Comment:
I don't understand `gosleep` but I might understand `turnOffOscillator`. Please rename to whatever is actually meant.